pandas/tests/dtypes/test_generic.py

```diff
@@ -38,3 +40,33 @@ def test_abc_types(self):
         assert isinstance(self.sparse_array, gt.ABCSparseArray)
         assert isinstance(self.categorical, gt.ABCCategorical)
         assert isinstance(pd.Period('2012', freq='A-DEC'), gt.ABCPeriod)
+
+
+class TestABCWarnings(object):
```

Comment:
I didn't see anything in the contributing guide about test style, so I was following the style of the code nearby.